### ISSUE TITLE
Purescript Rx example code

### ISSUE THREAD

**Issue by brucou** [anttih/purescript-rx#12]
Hi there, I know this is bleeding-edge but I was wondering if there is a code sample that can be reviewed to see how to bind Rxjs and purescript?

**Comment by anttih** [comment]
What kind of an example are you looking for specifically? A way to push events to an Observable (to listen for DOM events for example) or an example of how to setup a project with rxjs and purescript?

**Comment by brucou** [comment]
Some background on what I am trying to achieve. I grew quite knowledgeable about Rxjs and its innerworkings, at the same time that I grew quite frustrated about the untyped nature of javascript. I reviewed the alternative typing solutions and my conclusion was that none of them were good enough as of today. I believe you either have the benefits of a fully typed language or it is better not to incurr in the cost of changing language/build/testing/etc. Hence my investigation of purescript. Interop with javascript could be improved (the hoops to transform js function in curried functions for instance) but it is manageable/acceptable.

So now I want to write my js apps with purescript. Purescript language is more or less clear to me for an initial level, the thorny questions are interop with existing javascript libraries, in particular the effectful ones. This is where I feel the need to review examples. 

An ideal example for the whole rxjs community would be the flux challenge : https://github.com/staltz/flux-challenge. It has examples of implementation in several paradigms (Elm or ClojureScript, for example). It is a quite complete example (vs. the basic ones).

Now simple examples will also help advance my understanding (basically whatever available will vs. whatever is not available). So yes : 

- first of all, the setup ('importing' of Rxjs, typing 'instructions', etc.)
- then example of creating observables, and using rxjs operators
- then using effects (DOM reading and writing, effect handlers including DOM handlers)
- then maybe the same thing with subjects (a different beast than observables)

**Comment by anttih** [comment]
Sorry for the response lag. Here's one small example https://github.com/anttih/purescript-rx/blob/master/examples/Examples.purs

You should also read about FFI from the PureScript by example book: https://leanpub.com/purescript/read

I think one of the best ways to bundle up purescript and js code is the purescript-webpack plugin. Here's an example on how to use that: https://github.com/ethul/purescript-webpack-example

Basically you'd install rxjs with `npm install rxjs` and then bundle using webpack.

As for subjects, pruescript-rx doesn't yet provide and functions for working with Subjects. What you can do is create Subjects using FFI and coerce those Subjects into Observables.

**Comment by brucou** [comment]
Thank you very much for that information, it will take time to process all of it, but I appreciate it.
